src/HistoReg.cxx

```diff
@@ -1549,79 +1565,159 @@ int main(int argc, char* argv[])
+        GreedyParameters param_reslice_full;
+        GreedyParameters::SetToDefaults(param_reslice_full);
+
+        // Define dimension and mode
+        param_reslice_full.dim = 2;
+        param_reslice_full.mode = GreedyParameters::RESLICE;
+
+        // Define transformation to be applied
+        ResliceSpec Reslices_images_full;
+        vector <TransformSpec> Transformations_full;
+        TransformSpec TransformDiffFull, TransformAffFull;
+
+        TransformDiffFull.filename = PATH_big_warp;
+        TransformAffFull.filename = PATH_affine;
+
+        Transformations_full.push_back(TransformDiffFull);
+        Transformations_full.push_back(TransformAffFull);
+        param_reslice_full.reslice_param.transforms = Transformations_full;
+
+        // Define source, target, interpolation mode and output
+        Reslices_images_full.moving = PATH_new_source;
+        Reslices_images_full.interp = Interp;
+        Reslices_images_full.output = PATH_registered_image;
+
+        param_reslice_full.reslice_param.images.push_back(Reslices_images_full);
+        param_reslice_full.reslice_param.ref_image = PATH_new_target;
+
+        // Run greedy
+        GreedyRunner<2, double>::Run(param_reslice_full);
+        cout << "   Done." << '\n';
+        end_intermediate = chrono::system_clock::now();
+        duration = chrono::duration_cast<chrono::seconds>(end_intermediate - start_intermediate).count();
+        cout << "Apply transformation to original images took : " << duration << " seconds." << '\n';
+        // Converting to PNGs
+        if (Flag_PNGs == 1) {
+            start_intermediate = chrono::system_clock::now();
+
+            cout << "Converting images into PNGs..." << '\n';
+            string PATH_new_target_png = PATH_Output_PNGs_full_resolution + "/new_target.png";
+            command = c2d_executable + " -mcs " + PATH_new_target + " -foreach -type uchar -endfor -omc " + PATH_new_target_png;
+            system(command.c_str());
+            cout << "   Target done..." << '\n';
+
+            string PATH_new_source_png = PATH_Output_PNGs_full_resolution + "/new_source.png";
+            command = c2d_executable + " -mcs " + PATH_new_source + " -foreach -type uchar -endfor -omc " + PATH_new_source_png;
+            system(command.c_str());
+            cout << "   Source done..." << '\n';
+
+            string PATH_registered_image_png = PATH_Output_PNGs_full_resolution + "/registeredImage.png";
+            command = c2d_executable + " -mcs " + PATH_registered_image + " -foreach -type uchar -endfor -omc " + PATH_registered_image_png;
+            system(command.c_str());
+            cout << "   Registered image done." << '\n';
+
+            end_intermediate = chrono::system_clock::now();
+            duration = chrono::duration_cast<chrono::seconds> (end_intermediate - start_intermediate).count();
+            cout << "Converting result to PNGs images took : " << duration << " seconds." << '\n';
+        }
```

Comment:
If a reslicing image is NOT provided, go through the same steps as before. This could be much more optimized instead of two entirely independent branches, but this was the quickest way to do it and my CLion interpreter broke so I was flying in the dark.